modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp

```diff
@@ -298,13 +370,51 @@ void ISO15118_chargerImpl::ready() {
 
     controller = std::make_unique<iso15118::TbdController>(tbd_config, callbacks, setup_config);
 
+    // if the vas providers report their supported vas services before the controller exists,
+    // we need to update the controller with the supported vas services after instantiation
+    update_supported_vas_services();
+
     try {
         controller->loop();
     } catch (const std::exception& e) {
         EVLOG_error << e.what();
     }
 }
 
+void ISO15118_chargerImpl::update_supported_vas_services() {
+    iso15118::d20::SupportedVASs supported_vas_services;
+
+    for (const auto& provider_services : supported_vas_services_per_provider) {
+        for (const auto& service_id : provider_services) {
+            // Check for duplicate service IDs across all providers
+            if (std::find(supported_vas_services.begin(), supported_vas_services.end(), service_id) !=
+                supported_vas_services.end()) {
+                EVLOG_AND_THROW(
+                    Everest::EverestConfigError("Duplicate VAS service ID found: " + std::to_string(service_id)));
```

Comment:
The question is whether you should really throw an exception here and then restart the entire charger.
Perhaps only an error should appear in the log and the service ID and the incorrectly configured VAS module should be ignored

Reply:
That is true, will change that